modules/javafx.controls/src/main/java/javafx/scene/control/skin/ListViewSkin.java

```diff
@@ -205,6 +205,10 @@ public ListViewSkin(final ListView<T> control) {
         behavior.setOnSelectNextRow(() -> onSelectNextCell());
         behavior.setOnScrollPageDown(this::onScrollPageDown);
         behavior.setOnScrollPageUp(this::onScrollPageUp);
+        Boolean isRemoveKeyMappings = (Boolean) control.getProperties().get("removeKeyMappingsForComboBoxEditor");
+        if (isRemoveKeyMappings != null && isRemoveKeyMappings) {
+            behavior.removeKeyMappingsForComboBoxEditor();
+        }
```

Comment:
hmm .. I would expect the ListViewBehavior to install the per-context mappings (vs. the ListViewSkin removing those that don't fit)

- comboBoxListViewSkin sets the context property to indicate that the list is part of the drop-down
- ListViewBehavior installs (or not) the mappings that it needs in the context as appropriate (which would remove the need for the removeKeyMappings method to be public)
- ListViewSkin has nothing special to do

What do I overlook?

Reply:
>     * ComboBoxListViewSkin sets the context property to indicate that the list is part of the drop-down

Property is named such that it does not show relationship but only indicates the action that is needed. `ComboBox` already adds one such property named `selectFirstRowByDefault`. If we add a property to indicate that the ListView is part of ComboBox then we might need to remove the existing property as well, and combine the actions together under one property. So seems like naming the property to indicate actions is consistent.



> ListViewBehavior to install the per-context mappings (vs. the ListViewSkin removing those that don't fit)

Installing(Not installing) specific KeyMappings has one difference compared to removing few later that, changes for installing specific mappings would be spread: 1) The KeyMappings in `FocusTraversalInputMap` will have to be broken into two parts like in the first commit. 2) In `ListViewBehavior` the default(line 82) and vertical orientation(line 130) KeyMappings will have to be broken into separate calls. Since this change is specifically for ComboBox , the regular code can be left as is and this fix can be treated as special case and kept separate packed into single call.



>     * remove the need for the removeKeyMappingsForComboBoxEditor method to be public

Keeping `ListViewSkin` untouched and making `removeKeyMappingsForComboBoxEditor()` private seems good to me too. I clearly overlooked. I shall move the change from `ListViewSkin` to end of `ListViewBehavior` constructor.

Reply:
good that we agree on where to put it :) 

Will have to think about about the remove installed mappings vs. change existing code to not installing. Right now, feels a bit smelly .. but I see the constraints (and might agree :) Will come back!

Reply:
btw, using a key-value pair in properties is just fine with me (might have been unclear, sry) - nothing else to do short of changing api (which feels too heavy weight)

One nit-pit for checking for its existence, an alternative slightly simplified pattern might be

    if (Boolean.TRUE.equals(control.getProperties().get(....)) {
        removeMappings(...);
    }

Reply:
okay, I still favor the not-adding-if-not-needed approach - and it's not _that_ much of a change to existing code. Basically, in the constructor of ListViewBehavior

- in a first block, do not add the mappings that (currently) are removed again if the  combo flag is set
-in a second block,  add the mappings for a stand-alone listView to all inputMaps as needed

code snippets:

        // not needed if in combo popup
        //  addDefaultMapping(listViewInputMap, FocusTraversalInputMap.getFocusTraversalMappings());
        addDefaultMapping(listViewInputMap,
                // not needed if in combo popup
                // new KeyMapping(HOME, e -> selectFirstRow()),
                ...
               // add those that are always needed
        }

        // same for child input maps
        addDefaultMappings(verticalInputMap, ...

        // add those for stand-alone listView
        // handle mappings that are needed only in a stand-alone listView
        if (!Boolean.TRUE.equals(control.getProperties().get("removeKeyMappingsForComboBoxEditor"))) {
            addDefaultMapping(listViewInputMap, FocusTraversalInputMap.getFocusTraversalMappings());
            addDefaultMapping(listViewInputMap, 
                    new KeyMapping(HOME, e -> selectFirstRow()),
                    new KeyMapping(END, e -> selectLastRow()),
                    new KeyMapping(new KeyBinding(HOME).shift(), e -> selectAllToFirstRow()),
                    new KeyMapping(new KeyBinding(END).shift(), e -> selectAllToLastRow()),
                    new KeyMapping(new KeyBinding(A).shortcut(), e -> selectAll()),
                    new KeyMapping(new KeyBinding(HOME).shortcut(), e -> focusFirstRow()),
                    new KeyMapping(new KeyBinding(END).shortcut(), e -> focusLastRow())
            );

            addDefaultMapping(verticalListInputMap, 
                    new KeyMapping(new KeyBinding(HOME).shortcut().shift(), e -> discontinuousSelectAllToFirstRow()),
                    new KeyMapping(new KeyBinding(END).shortcut().shift(), e -> discontinuousSelectAllToLastRow())
            );
            
        }

Tests are passing as expected, and the [open issue JDK-8250807](https://bugs.openjdk.java.net/browse/JDK-8250807) doesn't stand in the way :)

As to testing, independent of which approach is chosen at the end: I would suggest to not only test the macroscopic behavior (look if key events on combo/listView have the expected effect) but also test the base layer, that is whether or not the mappings are not/added to the inputMaps. 
 
One open (maybe not-an) issue might be the handling of the horizontal map which currently is not touched (but probably will, once the deep removal of [JDK-8250807](https://bugs.openjdk.java.net/browse/JDK-8250807) is fixed) : while not very probable, client code might change the listView in the popup to be horizontal. What would be the expected UI in that case? Anyway, could be left open for the scope of this issue, IMO.

Reply:
> okay, I still favor the not-adding-if-not-needed approach

I have updated the PR to use this approach. I did not think that all of `FocusTraversalInputMap.getFocusTraversalMappings()`  can be removed. But looks like they can be removed without any side effect. This approach looks good.
Additionally, changed name of the property as `excludeKeyMappingsForComboBoxEditor` to reflect the action.


> whether or not the mappings are not/added to the inputMaps.

Added this test for both ComboBox and ListView.



> Anyway, could be left open for the scope of this issue, IMO.

I also think the same that it can be kept out of scope for now.